app/keepers.go

```diff
@@ -0,0 +1,326 @@
+		app.BlockedAddrs(),
+	)
+	app.BankKeeper = &bankKeeper
+
+	authzKeeper := authzkeeper.NewKeeper(
+		keys[authzkeeper.StoreKey],
+		appCodec,
+		bApp.MsgServiceRouter(),
+	)
+	app.AuthzKeeper = &authzKeeper
+
+	stakingKeeper := stakingkeeper.NewKeeper(
+		appCodec,
+		keys[stakingtypes.StoreKey],
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.GetSubspace(stakingtypes.ModuleName),
+	)
+	app.StakingKeeper = &stakingKeeper
+
+	distrKeeper := distrkeeper.NewKeeper(
+		appCodec, keys[distrtypes.StoreKey],
+		app.GetSubspace(distrtypes.ModuleName), app.AccountKeeper, app.BankKeeper,
+		&stakingKeeper, authtypes.FeeCollectorName, app.BlockedAddrs(),
+	)
+	app.DistrKeeper = &distrKeeper
+
+	slashingKeeper := slashingkeeper.NewKeeper(
+		appCodec, keys[slashingtypes.StoreKey], &stakingKeeper, app.GetSubspace(slashingtypes.ModuleName),
+	)
+	app.SlashingKeeper = &slashingKeeper
+
+	// Create IBC Keeper
+	app.IBCKeeper = ibckeeper.NewKeeper(
+		appCodec,
+		keys[ibchost.StoreKey],
+		app.GetSubspace(ibchost.ModuleName),
+		&stakingKeeper,
+		app.UpgradeKeeper,
+		app.ScopedIBCKeeper)
+
+	// Create Transfer Keepers
+	transferKeeper := ibctransferkeeper.NewKeeper(
+		appCodec, keys[ibctransfertypes.StoreKey], app.GetSubspace(ibctransfertypes.ModuleName),
+		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		app.AccountKeeper, app.BankKeeper, app.ScopedTransferKeeper,
+	)
+	app.TransferKeeper = &transferKeeper
+	app.transferModule = transfer.NewAppModule(*app.TransferKeeper)
+
+	// Create static IBC router, add transfer route, then set and seal it
+	ibcRouter := porttypes.NewRouter()
+	ibcRouter.AddRoute(ibctransfertypes.ModuleName, app.transferModule)
+	app.IBCKeeper.SetRouter(ibcRouter)
+
+	app.Bech32IBCKeeper = bech32ibckeeper.NewKeeper(
+		app.IBCKeeper.ChannelKeeper, appCodec, keys[bech32ibctypes.StoreKey],
+		app.TransferKeeper,
+	)
+
+	// TODO: Should we be passing this instead of bank in many places?
+	// Where do we want send coins to be cross-chain?
+	app.Bech32ICS20Keeper = bech32ics20keeper.NewKeeper(
+		app.IBCKeeper.ChannelKeeper,
+		app.BankKeeper, app.TransferKeeper,
+		app.Bech32IBCKeeper,
+		app.TransferKeeper,
+		appCodec,
+	)
+
+	// create evidence keeper with router
+	// If evidence needs to be handled for the app, set routes in router here and seal
+	app.EvidenceKeeper = evidencekeeper.NewKeeper(
+		appCodec, keys[evidencetypes.StoreKey], app.StakingKeeper, app.SlashingKeeper,
+	)
+
+	app.ClaimKeeper = claimkeeper.NewKeeper(
+		appCodec,
+		keys[claimtypes.StoreKey],
+		app.AccountKeeper,
+		app.BankKeeper, app.StakingKeeper, app.DistrKeeper)
+
+	gammKeeper := gammkeeper.NewKeeper(
+		appCodec, keys[gammtypes.StoreKey],
+		app.GetSubspace(gammtypes.ModuleName),
+		app.AccountKeeper, app.BankKeeper, app.DistrKeeper)
+	app.GAMMKeeper = &gammKeeper
+
+	app.LockupKeeper = lockupkeeper.NewKeeper(
+		appCodec, keys[lockuptypes.StoreKey],
+		// TODO: Visit why this needs to be deref'd
+		*app.AccountKeeper,
+		app.BankKeeper)
```

Comment:
Need to double check why Account Keeper has to be deref'd for LockupKepeer. Its odd

Reply:
Ah, I know why! It's because we're not using Keeper interface from expected_keeper as a parameter `NewKeeper` is taking in. Instead what NewKeeper is taking in as a parameter is authKeeper brought directly from cosmos-sdk. Issue needed for these!

Reply:
Ah awesome, thanks for finding that!